exporters/readers/s3_reader.py

```diff
@@ -228,88 +187,11 @@ def __init__(self, *args, **kwargs):
                                                 self.read_option('aws_access_key_id'),
                                                 self.read_option('aws_secret_access_key'))
         self.keys = self.keys_fetcher.pending_keys()
-        self.read_keys = []
-        self.current_key = None
-        self.last_block = 0
         self.logger.info('S3Reader has been initiated')
-        self.tmp_folder = tempfile.mkdtemp()
-        self.lines_reader = self.read_lines_from_keys()
 
     def get_read_streams(self):
         from exporters.bypasses.stream_bypass import Stream
         for key_name in self.keys:
             key = self.bucket.get_key(key_name)
             file_obj = urlopen(key.generate_url(S3_URL_EXPIRES_IN))
             yield Stream(file_obj, key_name, key.size)
```

Comment:
Yeah... one way of having retry here is having the get_read_streams return only stream metadata (file name, file size and any miscellaneous metadata the reader might need) and adding a `open_stream(stream_metadata)` function that can be called several times to re-open the stream if we can't read the whole thing. This would also help with restarting the export since we could store the stream metadata and then open the stream without having to call `get_read_streams` again.

I would like to implement that, but in a future pull request since this is not introducing that regression.